### PR TITLE
Greasemonkey: fix early GM_addStyle with fast sites.

### PR DESCRIPTION
**Pull request by toofar**
nemanjan00 reported this script not having any effect https://gist.github.com/nemanjan00/01537ec65cd672d1f125170c7807f400/raw/da850e49cc28bc9c94e1a3fb564939eaa2e0aa77/duckduckgo-deepdark.user.js

It turns out that the previous implementation of GM_addStyle was relying
on `document.onreadystatechange` when the script ran early enough that
there was no `head` element. That event wasn't getting fired for the
main frame of duckduckgo.com for whatever reason. Maybe using
`DOMContentLoaded` or something would have worked but I just copied the
fallback in the above linked script which seems to work just fine.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3792)

### CODE REVIEW

File: qutebrowser/javascript/greasemonkey_wrapper.js [hunk 1]
```diff
@@ -100,11 +100,8 @@
 
         const head = document.getElementsByTagName("head")[0];
```
Comment (jgkamat) [inline qutebrowser/javascript/greasemonkey_wrapper.js]: Not part of this PR, but can't

```
document.getElementsByTagName("head")[0];
```
be replaced with

```
document.head;
```

I could be understanding it wrong though...

Comment (toofar) [reply qutebrowser/javascript/greasemonkey_wrapper.js]: Eh, I just inherited that bit, and I am pretty sure it has been copied from somewhere. There is a similar greasemonkey shim in the chromium source somewhere.

>maybe it's coming from a child frame
Yeah I saw it being called twice for a `/post2` or some such url, but not the top level one.
